### PR TITLE
Fix various warnings

### PR DESCRIPTION
**Pull request by magopian**
refs #202

### CODE REVIEW

**Comment by magopian** [comment]
Regarding the warnings displayed in #202, especially the very last one:

```Warning: `getFieldDecorator` will override `value`, so please don't set `value` directly and use `setFieldsValue` to set it.```

This is displayed whenever you wrap a component in a `FormItem` with the [getFieldDecorator](https://ant.design/components/form/#getFieldDecorator(id,-options)-parameters).

This field decorator manages the value and initial value of the wrapped component, and thus warns about any "default value" that you set on the component itself (as it'll be overwritten by the `getFieldDecorator` utility).

I'm not sure how to fix this: I could remove the [default value](https://github.com/mozilla/delivery-console/blob/049fb3d3072caaa4753b86da2f43b516e30c0349/src/components/forms/IdenticonField.js#L19), but then if the `IdenticonField` component was used elsewhere without being wrapped in a `FormItem`,  then we'd miss the default value.

BTW, this warning should also be displayed for some other components like [ActionSelect](https://github.com/mozilla/delivery-console/blob/049fb3d3072caaa4753b86da2f43b516e30c0349/src/components/recipes/RecipeForm.js#L133), but the warning is "hidden" by the fact that the component is wrapped by `connect` (the ant code that detects if there's props related to `value` in the component can't find anything in the connect wrapper).

@rehandalal what do you think is the right way to go here? Any advice or insight?

File: src/components/forms/IdenticonField.js [hunk 1]
```diff
@@ -68,7 +67,6 @@ export default class IdenticonField extends React.PureComponent {
       index: newIndex,
       history: newHistory,
     });
-    this.props.onChange(next);
```
Comment (rehandalal) [inline src/components/forms/IdenticonField.js]: Removing this breaks the identicon picker in the form. You are no longer able to pick identicons.

This should become:
```js
const { onChange } = this.props; // Should go at top of function

if (onChange) {
  onChange(next);
}
```

File: src/components/forms/IdenticonField.js [hunk 2]
```diff
@@ -10,7 +10,6 @@ import ShieldIdenticon from 'console/components/common/ShieldIdenticon';
 export default class IdenticonField extends React.PureComponent {
   static propTypes = {
     disabled: PropTypes.bool,
-    onChange: PropTypes.func.isRequired,
```
Comment (rehandalal) [inline src/components/forms/IdenticonField.js]: you should just change this to:
```js
onChange: PropTypes.func,
```

**Comment by rehandalal** [comment]
To fix:
```
Warning: `getFieldDecorator` will override `value`, so please don't set `value` directly and use `setFieldsValue` to set it.
```

Remove the following from `defaultProps`:
```js
value: null
```

Then update the `render` method to have:
```jsx

        {value && (
          <Popover
            mouseEnterDelay={0.75}
            content={<ShieldIdenticon seed={value} size={256} />}
            placement="right"
          >
            <div className="shield-container">
              <ShieldIdenticon seed={value} />
            </div>
          </Popover>
        )}
```

**Comment by rehandalal** [comment]
Closing in favor of #280

**Pull request closed without merging**